src/thunderbots/software/ai/world/robot.cpp

```diff
@@ -152,15 +163,60 @@ AngularVelocity Robot::estimateAngularVelocityAtFutureTime(
     // angular velocity and does not necessarily reflect real-world behavior. Angular
     // velocity prediction should be improved as outlined in
     // https://github.com/UBC-Thunderbots/Software/issues/50
-    return angularVelocity_;
+    return angularVelocities_.back();
+}
+
+std::vector<Point> Robot::getPreviousPositions()
+{
+    std::vector<Point> retval{};
+    for (Point p : positions_)
+        retval.push_back(p);
+
+    return retval;
+}
```

Comment:
What was the rational behind not returning a circular buffer here?

Reply:
In the description of the ticket it said to return vectors so I didn't put too much thought into it. However in my opinion these functions get a snapshot of the history with no intention of changing or updating it so there would be no need for the circular buffer functionality so returning a more conventional vector made sense.

Reply:
fair enough